### PR TITLE
Feature/improve testing suite

### PR DESCRIPTION
**Pull request by mr-rooftop**

### CODE REVIEW

File: test/integration_test.py [hunk 1]
```diff
@@ -4,16 +4,17 @@
...
 from subprocess import Popen
 from pyrogram import Client as TelegramClient
 
 # Delete previous sessions for clean testing
-if os.path.exists("../storage/session.data"):
-    os.remove("../storage/session.data")
-
-# Start the Telegram Thornode Bot
-thornode_bot_process = Popen(['python3', 'thornode_bot.py'], cwd="../")
+#if os.path.exists("../storage/session.data"):
```
Comment (mbartn) [inline test/integration_test.py:13]: We shouldn't push any commented lines into master imo.

File: test/integration_test.py [hunk 2]
```diff
@@ -46,521 +47,579 @@
...
-
-def test_add_address(address, expected_response1, expected_response2):
-    telegram.send_message(BOT_ID, "/start")
-    time.sleep(3)
-    telegram.send_message(BOT_ID, "📡 MY NODES")
-    time.sleep(3)
-
-    click_button("1️⃣ ADD NODE")
+class ThornodeBot(unittest.TestCase):
+    thornode_bot_process = {}
```
Comment (mbartn) [inline test/integration_test.py:51]: I think we shouldn't assign dictonary here, more like `None`

File: test/integration_test.py [hunk 3]
```diff
@@ -46,521 +47,579 @@
...
+            os.remove("../storage/session.data")
 
-    assert first_response.text == expected_response1, "Expected '" + expected_response1 + "' but got '" + first_response.text + "'"
-    assert second_response_1.text == expected_response2 or second_response_2.text == expected_response2, \
-        "Expected '" + expected_response2 + "' but got '" + second_response_1.text + "' and '" + second_response_2.text + "'"
-    print("1️⃣ ADD NODE with " + address + " ✅")
-    print("------------------------")
+        # Authenticate Telegram Client of this testing suite
+        cls.telegram = TelegramClient(
+            open('telegram_session.string').read(),
```
Comment (mbartn) [inline test/integration_test.py:63]: is it possible that this file doesn't exist in that moment? Maybe if it doesn't we should display proper message

File: test/integration_test.py [hunk 4]
```diff
@@ -46,521 +47,579 @@
...
-        assert second_response_2.text.find("You're in the Admin Area - proceed with care") != -1, \
-            "YES button on restart confirmation does not go back to 🔑 ADMIN AREA"
-        assert second_response_2.reply_markup.inline_keyboard[0][0].text.find("second") != -1, \
-            "YES button on restart confirmation does not restart container"
-    else:
-        first_response.click("NO ❌")
+        response = next(self.telegram.iter_history(BOT_ID))
+
+        expected_response1 = 'Status of all THORNodes in the THORChain network:'
+        expected_response2 = 'Address: ' + VALID_ADDRESS
```
Comment (mbartn) [inline test/integration_test.py:317]: `VALID_ADDRESS` reference is actually unresolved

File: test/integration_test.py [hunk 5]
```diff
@@ -46,521 +47,579 @@
...
-    Click back button and assert TG shows what was shown before
-    """
-
-    click_button("⬅️ BACK")
+        response = next(self.telegram.iter_history(BOT_ID))
+        response.click(button)
+        time.sleep(3)
 
-    response = next(telegram.iter_history(BOT_ID))
+    def tor_to_rune(self, tor_string):
```
Comment (mbartn) [inline test/integration_test.py:425]: Why don't we use tor_to_rune from `helpers.py?`

File: test/integration_test.py [hunk 6]
```diff
@@ -46,521 +47,579 @@
...
+            new_value = statuses[new_value % 3]
+        else:
+            new_value += int(node_data_new[0][field])
+
+        node_data_new[0][field] = str(new_value)
+
+        with open('nodeaccounts.json', 'w') as json_write_file:
+            json.dump(node_data_new, json_write_file)
+
+        time.sleep(70)
```
Comment (mbartn) [inline test/integration_test.py:574]: these sleep() methods looks really too long for me, can we e.g. increase notification frequency while running tests?

File: test/integration_tests.py [hunk 7]
```diff
@@ -0,0 +1,765 @@
...
+
+from subprocess import Popen
+from pyrogram import Client as TelegramClient
+
+# Delete previous sessions for clean testing
+#if os.path.exists("../storage/session.data"):
+#    os.remove("../storage/session.data")
+#
+## Start the Telegram Thornode Bot
+#thornode_bot_process = Popen(['python3', 'thornode_bot.py'], cwd="../")
```
Comment (mbartn) [inline test/integration_tests.py:17]: as above

File: test/integration_tests.py [hunk 8]
```diff
@@ -0,0 +1,765 @@
...
+
+"""
+######################################################################################################################################################
+TEST CASES
+######################################################################################################################################################
+"""
+
+
+class ThornodeBot(unittest.TestCase):
+    thornode_bot_process = {}
```
Comment (mbartn) [inline test/integration_tests.py:49]: as above

**Review by mbartn — CHANGES_REQUESTED** [review]
tests look much cleaner right now! Nice work

File: test/integration_test.py (hunk 6 above)
Comment (mr-rooftop) [reply test/integration_test.py]: I say it as Obama would: "YES WE CAN"

**Review by mbartn — APPROVED** [review]

**Pull request merged**